controllers/discovery/initDiscoveryTopicData.js

```diff
@@ -9,6 +9,7 @@ const {sequelize} = require('../../databases/models');
  */
 const InitDiscoveryTopicData = async (req, res) => {
   let {limit = 10, page = 0} = req.query;
+  page = parseInt(page);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `parseInt` function can return `NaN` if the input is not a number. This could lead to unexpected behavior in subsequent code that uses the `page` variable. It would be better to handle this case explicitly.
```diff
-   page = parseInt(page);
+   page = Number.isNaN(parseInt(page)) ? 0 : parseInt(page);
```
This change ensures that if `page` cannot be parsed into an integer, it will default to `0`.


<!-- This is an auto-generated comment by OSS CodeRabbit -->